playbooks/acs.yml

```
@@ -1,4 +1,14 @@
 ---
+- name: Check if OS distribution is supported
+  hosts: all:!external
+  gather_facts: yes
+  tasks:
+  - name: compare host OS with supported matrix
+    vars:
+      os_versions: "{{ supported_os[ansible_distribution].versions | default([]) }}"
+    ansible.builtin.fail:
+      msg: "{{ ansible_distribution }} {{ ansible_distribution_major_version }} is not supported OS"
+    when: skip_os_test | default(false) and ansible_distribution_major_version | int not in os_versions
```

that logic is broken as `skip_os_test=true` would result in the task being executed.

```suggestion
    when:
    - not skip_os_test | default(false) | bool
    - ansible_distribution_major_version | int not in os_versions
```